src/kernel/src/dev/gc.rs

```diff
@@ -35,15 +37,29 @@ impl DeviceDriver for Gc {
         &self,
         _: &Arc<CharacterDevice>,
         cmd: IoCmd,
-        _: Option<&VThread>,
+        td: Option<&VThread>,
     ) -> Result<(), Box<dyn Errno>> {
+        if self.suspended {
+            todo!("gc suspended")
+        }
+
+        let td = td.unwrap();
+
+        let gc_check_passed = td.proc().cred().unk_gc_check();
```

Comment:
I saw there is a call to a function after this.

Reply:
That's devfs_get_cdevpriv (a FreeBSD function). I added a todo comment for now.